sklearn/preprocessing/_discretization.py

```diff
@@ -136,6 +143,13 @@ def fit(self, X, y=None):
         self
         """
         X = check_array(X, dtype='numeric')
+        n_samples, n_features = X.shape
+
+        if n_samples > self.subsample:
+            subsample_idx = np.random_state.choice(n_samples,
```

Comment:
Please avoid using the singleton `np.random_state` from numpy and create your own instance in fit as done in `QuantileTransformer`.

Reply:
Done